tests/algorithms/test_preference_comparisons.py

```diff
@@ -918,3 +920,141 @@ def test_agent_trainer_sample_image_observations(rng):
         trajectory.obs.shape[1:] == venv.observation_space.shape
         for trajectory in trajectories
     )
+
+
+class ActionIsRewardEnv(gym.Env):
+    """Two step environment where the reward is the sum of actions."""
+
+    def __init__(self):
+        """Initialize environment."""
+        super(ActionIsRewardEnv, self).__init__()
+        self.action_space = spaces.Discrete(50)
+        self.observation_space = gym.spaces.Box(np.array([0]), np.array([1]))
+        self.reward = 0.0
+        self.steps = 0
+
+    def step(self, action):
+        obs = np.array([0])
+        reward = action
+        # Some algorithms expect trajectories at least two steps long,
+        # so we allow two steps to be taken.
+        done = self.steps > 0
+        info = {}
+        self.steps += 1
+        return obs, reward, done, info
+
+    def reset(self):
+        self.steps = 0.0
+        return np.array([0])
+
+
+@pytest.fixture
+def action_is_reward_venv(rng):
+    return DummyVecEnv(
+        [partial(ActionIsRewardEnv)],
+    )
+
+
+@pytest.fixture
+def action_is_reward_agent(action_is_reward_venv, rng):
+    return stable_baselines3.PPO(
+        "MlpPolicy",
+        action_is_reward_venv,
+        n_epochs=1,
+        batch_size=2,
+        n_steps=10,
+    )
+
+
+@pytest.fixture
+def action_is_reward_trainer(
```

Comment:
Fair point, I went with just splitting it into two separate helper functions because I feel that using the closure is getting a little fancy to follow.

Thoughts on it now?